enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/discovery/DnsResolutionResolverTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.discovery;
+
+import org.junit.Test;
+
+import java.util.Collection;
+
+import org.neo4j.helpers.AdvertisedSocketAddress;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.junit.Assert.assertThat;
+
+public class DnsResolutionResolverTest
+{
+
+    private DnsResolutionResolver resolver = new DnsResolutionResolver();
+
+    @Test
+    public void shouldResolveDNSToIPs()
+    {
+        Collection<AdvertisedSocketAddress> resolve =
+                resolver.resolve( new AdvertisedSocketAddress( "google.com", 80 ) );
+        assertThat( resolve.size(), greaterThanOrEqualTo( 1 ) );
+    }
+
+    @Test
+    public void shouldResolveLocalhostIPDNSToIPs()
+    {
+        Collection<AdvertisedSocketAddress> resolve =
+                resolver.resolve( new AdvertisedSocketAddress( "127.0.0.1", 80 ) );
+        assertThat( resolve.size(), greaterThanOrEqualTo( 1 ) );
+    }
+
+    @Test
+    public void shouldResolveLocalhostDNSToIPs()
+    {
+        Collection<AdvertisedSocketAddress> resolve = resolver.resolve( new AdvertisedSocketAddress( "localhost", 80 ) );
+        assertThat( resolve.size(), greaterThanOrEqualTo( 1 ) );
+    }
+
+    @Test
+    public void shouldResolveIPv6LocalhostDNSToIPs()
+    {
+        Collection<AdvertisedSocketAddress> resolve = resolver.resolve( new AdvertisedSocketAddress( "[::1]", 80 ) );
+        assertThat( resolve.size(), greaterThanOrEqualTo( 1 ) );
+    }
+
+    @Test
+    public void shouldExceptionAndGiveDefaultValue()
```

Comment:
Sorry for the late reply - I can change it to being wrapped in a `RuntimeException(e)` and thrown back up if you like? That is probably the best solution anyway come to think of it.